docs/learning/ECSqlReference/InstanceQuery.md

```diff
@@ -0,0 +1,285 @@
+
+```json
+{
+  "ECInstanceId": "0x8000000014c",
+  "ECClassId": "0x710",
+  "Model": {
+    "Id": "0x80000000003",
+    "RelECClassId": "0x51"
+  },
+  "LastMod": "2020-09-13T21:03:39.281Z",
+  "CodeSpec": {
+    "Id": "0x1",
+    "RelECClassId": "0x59"
+  },
+  "CodeScope": {
+    "Id": "0x1",
+    "RelECClassId": "0x5b"
+  },
+  "UserLabel": "Computer",
+  "Category": {
+    "Id": "0x70000000034",
+    "RelECClassId": "0xa8"
+  },
+  "InSpatialIndex": true,
+  "Origin": {
+    "X": -20.17197015358312,
+    "Y": -12.999908317386943,
+    "Z": -5.363399999999998
+  },
+  "Yaw": -9.610521879999869,
+  "Pitch": 0,
+  "Roll": 0,
+  "BBoxLow": {
+    "X": -0.2844601562499974,
+    "Y": -0.34431570637657166,
+    "Z": -0.00034867627660684075
+  },
+  "BBoxHigh": {
+    "X": 0.4287276153476725,
+    "Y": 0.0297172168743558,
+    "Z": 0.5207000000000108
+  },
+  "GeometryStream": "encoding=base64;Ug==",
+  "TypeDefinition": {
+    "Id": "0x80000000145",
+    "RelECClassId": "0xcc"
+  },
+  "TypeId": "382002",
+  "RevitId": "381840",
+  "Timestamp": "2020-09-10T13:36:41.000",
+  "LastModifier": "kiran.patkar",
+  "ELEM_TYPE_PARAM": "Computer",
+  "ELEM_CATEGORY_PARAM": "Specialty Equipment",
+  "FAMILY_LEVEL_PARAM": "B1-CONCOURSE",
+  "Asset_Tag": "COMPUTER 005"
+}
+```
+
+Take a property `Asset_Tag` which might be property that exist on some instance of derived hierarchy of `bis.Element` and we like to find any instance of `Bis.Element` where `$->Asset_Tag ='COMPUTER 005'`
+
+```sql
+    SELECT [ECInstanceId] FROM [BisCore].[Element] WHERE $->[Asset_Tag] ='COMPUTER 005'
+```
+
+| ECInstanceId  |
+| ------------- |
+| 0x8000000014c |
+
+Similarly we can read any set of properties and also filter by them
+
+```sql
+    SELECT $->[RevitId], $->[LastModifier]  FROM [BisCore].[Element] WHERE $->[Asset_Tag] ='COMPUTER 005'
+```
+
+| $ -> RevitId | $ -> LastModifier |
+| ------------ | ----------------- |
+| 381840       | kiran.patkar      |
+
+ECSql will apply a property filter on selected rows such that those instances which has at least one property out of set of instance property must exists. This improve performance.
+
+```sql
+    SELECT $->[ThisPropertyDoesNotExists] from [BisCore].[Element];
+```
+
+If `ThisPropertyDoesNotExists` does not exists in `Bis.Element` derived hierarchy then no row will be returned. ECSql filter only include rows that must have at least one instance property. If any instance does not have any instance property requested then it will will be skipped.
+
+## Accessing composite properties like `NavigationProperty`, `Point2d`, `Point3d` or `Struct`s
+
+Only top level instance property can be accessed via `$-><prop>` syntax. Doing something like `$->Model.Id` will not not work as of now. It might be supported in future but as of now any access-string within a composite property is not supported, if its the only property selected then zero row will be returned.
```

Comment:
```suggestion
Only top level instance properties can be accessed via `$-><prop>` syntax. Doing something like `$->Model.Id` will not work. It might be supported in future but as of now any access-string within a composite property is not supported: if it is the only property selected then zero row will be returned.
```